src/components/Header/index.tsx

```diff
@@ -0,0 +1,252 @@
+import { faBars } from '@fortawesome/free-solid-svg-icons/faBars';
+import { faTimes } from '@fortawesome/free-solid-svg-icons/faTimes';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { BoxProps } from 'grommet';
+import React from 'react';
+import styled, { css } from 'styled-components';
+import { invertBreakpoint } from '../../utils';
+import { Box } from '../Box';
+import Button from '../Button';
+import Container from '../Container';
+import Fixed from '../Fixed';
+import { Flex } from '../Flex';
+import Img from '../Img';
+import Link from '../Link';
+import Txt from '../Txt';
+import { NavDropDown } from './NavDropDown';
+
+const NavigationItem = styled(Box)`
+	margin: 0;
+
+	a {
+		margin: 0 !important;
+		color: ${(props) => props.theme.colors.text.main};
+		display: block !important;
+		padding-top: 12px !important;
+		padding-bottom: 10px !important;
+		font-size: initial;
+
+		&:hover:before {
+			border: none;
+		}
+	}
+
+	svg {
+		float: right;
+		transform: translateY(4px);
+
+		path {
+			fill: ${(props) => props.theme.colors.text.main};
+		}
+	}
+`;
+interface HeaderProps extends BoxProps {
+	logo?: string;
+	logoAltText?: string;
+	rootRoute?: string;
+	routes: Route[];
+	actions?: React.ReactElement;
+}
+
+interface Route {
+	title: string;
+	showDividers?: boolean;
+	path?: string;
+	blank?: boolean;
+	routes: Route[];
+}
+const getNavigationRoutes = (routes: Route[]) => {
+	return routes.map((route, index) => {
+		if (route['routes']) {
+			return (
+				<NavigationItem pl={[0, 0, 3, 3]} key={index} mx={15}>
+					<NavDropDown title={route.title} showDividers={route.showDividers}>
+						{getNavigationRoutes(route.routes)}
+					</NavDropDown>
+				</NavigationItem>
+			);
+		}
+
+		return (
+			<NavigationItem pl={[0, 0, 3, 3]} key={index} mx={[2, 2, 15]}>
+				<Link
+					blank={route.blank || false}
+					href={route.path}
+					style={{
```

Comment:
Let's use styled system props (Link should support both of these, if it doesn't then it's a bug that needs fixing)